### ISSUE TITLE
failing to crack hash with john

### ISSUE THREAD

**Issue by kolet** [jmmcatee/cracklord#111]
hey, first thanks alot for all the works u guys are doing for all of us.

this is my johndic.conf
___________________________________________________

```
# For additional information see https://github.com/jmmcatee/cracklord/wiki/johntheripper-Dictionary

[Basic]
# Change the following line below to be where hashcat is installed.
# Note that resourced will, if you used a package, run as cracklord-user
# Depending on the drivers, hashcat sometimes needs to run as a specific user
binPath=/mnt/JohnTheRipper/run/john

# Where you would like to store all of your working files
# SET PERMISSIONS, THIS WILL CONTAIN HASHES!
workingdir=/var/cracklord/

# If you need to have additional arguments added to john, just put them here
arguments=

# List out all of the dictionaries you want to have available, one per line,
# The name on the left will appear to users, on the right should be the full
# path to the file.
[Dictionaries]
En=/home/test/Documents/English.dic
```

____________________________________________________

as u can see i added my own wordlist (English.dic)
and changed the path for john.

i added my resource server and put a new job,

type of atack= dahua
dic =En
rules, = i tried none,singel,all,wordlist and empty

and the hash itself
the hash is 6QNMIQGe which is admin, to crack it with john u need to add $dahua$ in the beging so what i've put was $dahua$6QNMIQGe and 6QNMIQGe to be sure whats the input that need to be.

i've put in the dictory admin inside the dic (English.dic),

after couple of second in the Q, its say the job is done but its 0/2 which mean it did not work...

would apprciate some help with this,

thanks!

Update 1:

this is resourced.conf

```
# For additional information and help on this file, see https://github.com/jmmcatee/cracklord/wiki/Configuration-Files#resource
[General]
# This next three directives (CACertFile, CertFile, and KeyFile) will be used
# to help validate the certificate and authenticate the resource server with the
# queue server and to encrypt communications between the resource and queue

# The full path to the certificate authority public key used on the Queue server.
CACertFile=/etc/cracklord/ssl/cracklord_ca.pem
# The full path to the public certificate for the resource
CertFile=/etc/cracklord/ssl/resourced.crt
# The full path to the private key for the resource
KeyFile=/etc/cracklord/ssl/resourced.key

# The IP address and port the resource server will listen on for connections from the queue
BindIP=0.0.0.0
BindPort=9443

# The file where logs will be written to
LogFile=/var/log/cracklord/resourced.log
# The level of messages for logs (Debug, Info, Warn, Error, Fatal, Panic)
LogLevel=Info

[Plugins]
# For each plugin you want to run on this resource, uncomment the lines below
# and make sure the files exist, as this is just a default.
hashcat=/etc/cracklord/plugins/hashcat.conf
nmap=/etc/cracklord/plugins/nmap.conf
johndict=/etc/cracklord/plugins/johndict.conf
```

UPDATE 2:
when checking /var/cracklord 
it shows every id job that i add, when viewing the .pot files, it show the password which mean it cracked the hash , but the its still showing 0/1

i was thinking it might be a permission problem because when i ls -l
```
/cracklord# ls -l
total 16
drwx------ 2 root root 4096  apr 10 17:05 f0e67416-133f-42d2-b0b6-275e6fb58820
-rw-r--r-- 1 root root 7152  apr 10 16:52 queue.state
drwxrwxr-x 6 root root 4096  apr 10 13:33 www
```

so i changed the cracklord group as root, changed the gid and uid to 0 (so cracklord have full priv)

same issue occurred. still 0/1

even when deleting the jobs from the web application the jobs id are still there.

**Comment by jmmcatee** [comment]
This is likely an issue I've fixed in the cracklord code already when a job is quick or finishes at a weird time based on the multiple threads. I have not looked at the John code for a while and it likely needs some love. I will add it to our Milestone one work to get John up to snuff. Sorry for the issue. Does this example you provided run very quickly?

**Comment by klingsor83** [comment]
I have the same issue. If I create the job with cracklord, it generates the directory with the log, pot and hashes files inside but it never finishes with the cracked password.

I tried to put in the terminal the command executed by cracklord (deleting previously the .log and .pot) and it finished correctly in 2 seconds.

Also, I tried to use the plugin with more complex passwords and others algorithms and it still always 0/1. I can't get a cracked password with JTR and Cracklord :(
